client/main.ts

```diff
@@ -28,24 +28,29 @@ async function start() {
 		await loadFromDB(page.thread)
 	}
 
-	// Check for legacy options and remap
-	const oldNowPlaying = localStorage.getItem("nowPlaying")
-	switch (oldNowPlaying) {
-	case "false":
-	case "none":
-		localStorage.setItem("nowPlaying", "" + (0 ^ 0));
-		break;
-	case "true":
-	case "r/a/dio":
-		localStorage.setItem("nowPlaying", "" + (0 ^ 1));
-		break;
-	case "eden":
-		localStorage.setItem("nowPlaying", "" + (0 ^ 2));
-		break;
-	case "both":
-		localStorage.setItem("nowPlaying", "" + (0 ^ 1 ^ 2));
-		break;
+	// Migrate nowPlaying options
+	const nowPlaying = localStorage.getItem("nowPlaying")
+	switch (nowPlaying) {
+		case "true":
+		case "r/a/dio":
+			localStorage.setItem("radio", "true")
+			break
+		case "eden":
+			localStorage.setItem("eden", "true")
+			break
+		case "both":
+			localStorage.setItem("radio", "true")
+			localStorage.setItem("eden", "true")
+			break
+		default:
+			const flags = parseInt(nowPlaying, 10)
```

Comment:
It's amazing how this still works with `nowPlaying=null`.

Reply:
NaN is converted to 0 in bitwise contexts. That migration should only run if nowPlaying exists however. Good catch.